### PR TITLE
Delete IE from datastore

### PR DESCRIPTION
**Pull request by mdittmer**
IE data have already been backed up. This code should not be needed again,
but if IE data are unintentionally imported in the future it will come in
handy.

### CODE REVIEW

File: main/datastore_delete_ie.es6.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    foam.lookup('org.chromium.apis.web.DatastoreContainer').create({
+      gcloudAuthEmail: credentials.client_email,
+      gcloudAuthPrivateKey: credentials.private_key,
+      gcloudProjectId: credentials.project_id,
+      logger: logger,
+    }).ctx;
+
+const E = foam.lookup('foam.mlang.ExpressionsSingleton').create();
+
+
```
Comment (arobins) [inline main/datastore_delete_ie.es6.js]: Remove extra blank lines

File: main/datastore_delete_ie.es6.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+
+const ApiVelocityData = foam.lookup('org.chromium.apis.web.ApiVelocityData');
+const apiVelocityDataPromise =
+    datastoreCtx.apiVelocityDAO.where(E.EQ(ApiVelocityData.BROWSER_NAME, 'IE'))
+    .removeAll()
```
Comment (arobins) [inline main/datastore_delete_ie.es6.js]: This indentation looks weird.

File: main/datastore_delete_ie.es6.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+const Release = foam.lookup('org.chromium.apis.web.Release');
+const ReleaseWebInterfaceJunction =
+    foam.lookup('org.chromium.apis.web.ReleaseWebInterfaceJunction');
+const releaseApiJunctionPromise =
+    datastoreCtx.releaseDAO.where(E.EQ(Release.BROWSER_NAME, 'IE')).select()
+    .then(releaseSink => Promise.all(releaseSink.array.map(release => {
+      let promises = [];
+      datastoreCtx.releaseWebInterfaceJunctionDAO.where(
+          E.GT(ReleaseWebInterfaceJunction.SOURCE_ID, release.releaseKey))
```
Comment (arobins) [inline main/datastore_delete_ie.es6.js]: Why is this GT?

File: main/datastore_delete_ie.es6.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+          .select(foam.lookup('foam.dao.QuickSink').create({
+            putFn: junction => promises.push(
+                datastoreCtx.releaseWebInterfaceJunctionDAO.remove(junction))
+          })).then(() => Promise.all(promises));
+    }))).then(() => logger.info(`Removed all releases <--> API junctions with
+                                    browserName="IE"`));
```
Comment (arobins) [inline main/datastore_delete_ie.es6.js]: Is there any need/desire to clean up APIs that are orphaned as a result of this cleanup?

File: main/datastore_delete_ie.es6.js (hunk 3 above)
Comment (mdittmer) [inline main/datastore_delete_ie.es6.js]: It shouldn't be. Long story as to exactly how this is a copy/paste error. Will fix.

File: main/datastore_delete_ie.es6.js (hunk 2 above)
Comment (mdittmer) [inline main/datastore_delete_ie.es6.js]: +4?

File: main/datastore_delete_ie.es6.js (hunk 4 above)
Comment (mdittmer) [inline main/datastore_delete_ie.es6.js]: There is, indeed.

File: main/datastore_delete_ie.es6.js (hunk 2 above)
Comment (arobins) [inline main/datastore_delete_ie.es6.js]: I think +4 from the start of datastoreCtx is the right thing to do.

File: main/datastore_delete_ie.es6.js [hunk 5]
```diff
@@ -56,23 +53,43 @@ const browserMetricsDataPromise =
...
+            putFn: junction => {
+              logger.info(`Removing Release/API junction: ${junction.id}`);
+              promises.push(
+                  datastoreCtx.releaseWebInterfaceJunctionDAO.remove(junction));
+            }
           })).then(() => Promise.all(promises));
     }))).then(() => logger.info(`Removed all releases <--> API junctions with
-                                    browserName="IE"`));
+                                    browserName="IE"`))
+    // Remove interfaces that ar IE-only.
```
Comment (arobins) [inline main/datastore_delete_ie.es6.js]: nit: typo in are

**Pull request merged**